### PR TITLE
OAuth Consent page URL generation set to comply with tenant qualification

### PR DESCRIPTION
**Pull request by ashendes**
**Related issues:**
[product-is/issues/7993](https://github.com/wso2/product-is/issues/7993)

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java [hunk 1]
```diff
@@ -610,9 +610,9 @@ public static String getUserConsentURL(OAuth2Parameters params, String loggedInU
             }
 
             if (isOIDC) {
-                consentPage = OAuth2Util.OAuthURL.getOIDCConsentPageUrl();
+                consentPage = IdentityUtil.resolveURL(OAuth2Util.OAuthURL.getOIDCConsentPageUrl(), true, true);
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java]: Should we do this?
With changes done by @ChamathNS, OAuth2Util.OAuthURL.getOIDCConsentPageUrl() should return the tenant qualified URLs, isn't it?

File: components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java [hunk 2]
```diff
@@ -2483,7 +2483,8 @@ private Response handleAuthFlowThroughFramework(OAuthMessage oAuthMessage, Strin
                 if (attribute == AuthenticatorFlowStatus.INCOMPLETE) {
 
                     if (responseWrapper.isRedirect()) {
-                        oAuthMessage.getResponse().sendRedirect(responseWrapper.getRedirectURL());
+                        return Response.status(HttpServletResponse.SC_FOUND).location(
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java]: What's the reason for this change?

Comment (malithie) [reply components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java:2486]: @ashendes , did we check removing this.
The first explation was this was added as this "returns 204 instead of 302 when servlet forwards are done at the rest dispatcher"

File: components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java (hunk 1 above)
Comment (malithie) [reply components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java:613]: This should be fixed in OAuth2Util.OAuthURL such that URL is resolved always an tenant url supported mode, until @ChamathNS get that fixed from the API
